tree/arbo/tree.go

```diff
@@ -620,6 +625,8 @@ func (t *Tree) SetRootWithTx(wTx db.WriteTx, root []byte) error {
 // The provided root must be a valid existing intermediate node in the tree.
 // The list of roots for a level can be obtained using tree.RootsFromLevel().
 func (t *Tree) Snapshot(fromRoot []byte) (*Tree, error) {
+	t.Lock()
+	defer t.Unlock()
```

Comment:
I assume these were added while you were debugging the pebble panic, and should now be removed?

Reply:
not sure what happened here - I still think these added locks/unlocks were only there while you debugged pebble

Reply:
Yes, you are right. But while I was reading the Arbo code, I found some missing locks. Which might not be critical, but I prefer to make it secure.